dali/kernels/erase/erase_gpu_test.cu

```diff
@@ -0,0 +1,390 @@
+  }
+  {
+    ivec<4> region_dims = {2, 2, 32, 32};
+    ivec<4> sample_dims = {16, 8, 32, 64};
+    ivec<4> expected_cover = {8, 4, 1, 2};
+    verify_regions(region_dims, sample_dims, expected_cover);
+  }
+  {
+    ivec<4> region_dims = {2, 2, 32, 32};
+    ivec<4> sample_dims = {16, 8, 64, 32};
+    ivec<4> expected_cover = {8, 4, 2, 1};
+    verify_regions(region_dims, sample_dims, expected_cover);
+  }
+  {
+    ivec<4> region_dims = {2, 2, 32, 32};
+    ivec<4> sample_dims = {16, 8, 32, 32};
+    ivec<4> expected_cover = {8, 4, 1, 1};
+    verify_regions(region_dims, sample_dims, expected_cover);
+  }
+}
+
+template <typename T, int ndim, int channel_dim = -1>
+struct EraseGpuKernelTest :
+    public testing::TestWithParam<std::tuple<
+      int,  // num erase regions
+      int,  // region generation scheme
+      int,  // default fill or full channel fill
+      TensorShape<ndim>>> {  // shape of input
+  void SetUp() override {
+    auto params = this->GetParam();
+    num_erase_regions_ = std::get<0>(params);
+    region_generation_ = std::get<1>(params);
+    full_fill_ = std::get<2>(params);
+    shape_ = std::get<3>(params);
+    test_shape = uniform_list_shape<ndim>(batch_size, shape_);
+
+    input.reshape(test_shape);
+    output.reshape(test_shape);
+    baseline.reshape(test_shape);
+    auto cpu_input_view = input.cpu();
+    SequentialFill(cpu_input_view);
+    if (full_fill_) {
+      fill_values.resize(shape_[channel_dim]);
+      int value = 0;
+      for (auto &elem : fill_values) {
+        elem = value++;
+      }
+    } else {
+      fill_values.resize(1);
+      fill_values[0] = 0;
+    }
+  }
+
+  void RunTest() {
+    if (region_generation_ == 0) {
+      std::cerr << ">> No cover" << std::endl;
+    } else if (region_generation_ == 1) {
+      std::cerr << ">> Full cover" << std::endl;
+    } else if (region_generation_ == 2) {
+      std::cerr << ">> Random cover of size: " << num_erase_regions_ << std::endl;
+    }
+    EraseGpu<T, ndim, channel_dim> kernel;
+    KernelContext ctx;
+    ctx.gpu.stream = 0;  // I need to know about this, it's really obfuscated
+    // we really should have a nice way of creating the contexts.
+
+    CreateRegions();
+
+    auto regions_gpu = regions.gpu();
+
+    auto in_view = input.gpu();
+
+    auto req = kernel.Setup(ctx, in_view, regions_gpu, fill_values);
+
+    auto out_view = output.gpu();
+
+     // AND THIS IS NO BETTER:
+    ScratchpadAllocator scratch_alloc;
+    scratch_alloc.Reserve(req.scratch_sizes);
+    auto scratchpad = scratch_alloc.GetScratchpad();
+    ctx.scratchpad = &scratchpad;
+
+    // auto start = CUDAEvent::CreateWithFlags(0);
+    // auto end =   CUDAEvent::CreateWithFlags(0);
+    // cudaEventRecord(start);
+
+    kernel.Run(ctx, out_view, in_view, regions_gpu, fill_values);
+
+    // cudaEventRecord(end);
+    CUDA_CALL(cudaDeviceSynchronize());
+    CUDA_CALL(cudaGetLastError());
+
+    // float t = 0;
```

Comment:
Maybe print this in debug or always (as we do in some reduce tests)?

Reply:
No strong desire to keep it, so removed everywhere.